### ISSUE TITLE
Remix 1.14.0 flat route folders with _index include everything inside as child routes

### ISSUE THREAD

**Issue by kevinbailey25** [remix-run/remix#5672]
### What version of Remix are you using?

1.14.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Create a fresh remix site via `npx create-remix@latest` using all defaults.
Enable flat routes in `remix.config.js`

Rename `index.tsx` to `_index.tsx` to use conventions for flat routes.

Add a folder flat route `some-folder._index/route.tsx`
Add a basic file in your `some-folder._index` like `utils.ts` with any exports

`npm run dev` or even `npx remix routes` and you'll get an error:
```
Error: Child routes are not allowed in index routes. Please remove child routes of some-folder._index
```

### Expected Behavior

I can have a flat route folder ending in `_index` that allows me to co-locate files within it and doesn't count those as child routes.

### Actual Behavior

```
Error: Child routes are not allowed in index routes. Please remove child routes of some-folder._index
```

**Comment by kevinbailey25** [comment]
Windows 10

**Comment by michaelgmcd** [comment]
Yea, this is a windows issue. I am working on a fix now. The issue is that this line returns `false` on mac and `true` on MacOS:
 https://github.com/remix-run/remix/blob/main/packages/remix-dev/config/flat-routes.ts#L382

I have a failing test in my fork: https://github.com/michaelgmcd/remix/commit/8e61f7282f9c8738134ecb62587381168421eba7

**Comment by kevinbailey25** [comment]
Thanks so much. I spent more time than I'd like to admit digging through my code and remix's code with a coworker to try and figure this out.

**Comment by bluefire2121** [comment]
Thank you for fixing this issue. Windows user here and this is failing in 1.14.2

**Comment by ngbrown** [comment]
I was getting this error on `v1.14.0` and `v0.0.0-nightly-bf9d0a9-20230314` fixed it.

**Comment by kevinbailey25** [comment]
This is fixed with `v1.15.0`
